fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -220,6 +220,8 @@ public static Object toClassWithRealEquals(@Nullable Object obj) {
             return obj;
         } else if (obj instanceof List) {
             return obj;
+        } else if (obj instanceof Message) {
```

Comment:
This should not be in here as the class does not really have an `equals()` that we like. I changed the callers on my local build to special-case `Message` to always call the `MessageHelper` method.